core/src/main/java/org/infinispan/protostream/impl/StringUtil.java

```diff
@@ -0,0 +1,155 @@
+package org.infinispan.protostream.impl;
+
+import static java.lang.invoke.MethodType.methodType;
+
+import java.lang.invoke.CallSite;
+import java.lang.invoke.LambdaMetafactory;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.util.function.Function;
+import java.util.function.ToIntFunction;
+
+import sun.misc.Unsafe;
+
+class StringUtil {
+
+   interface Utf8Helper {
+      boolean providesLatin1Bytes();
+      byte[] getBytes(String s);
+   }
+
+   private static final Log log = Log.LogFactory.getLog(ProtoStreamReaderImpl.class);
+   private static final Byte LATIN1 = (byte) 0;
+
+   static volatile Utf8Helper INSTANCE;
+
+   static Utf8Helper getUtf8Helper() {
+      if (INSTANCE == null) {
+         // Attempt to use IMPL_LOOKUP so that we utilise a Lambda - Most efficient
+         try {
+            INSTANCE = createLambdaHelper();
```

Comment:
I still question if this is valuable especially once Java 25 comes out, I am guessing many will _try_ to be on that version.

Reply:
Maybe we should add these to the commons all/jdk22 to better separate and not have to worry about checking for Unsafe? Then this code can also use `UnsafeHolder` to not have to repeat the code.